File: mediq-backend/src/main/java/com/mediq/service/PrescriptionService.java
package com.mediq.service;

import com.mediq.constants.PrescriptionStatus;
import com.mediq.dto.CreatePrescriptionRequest;
import com.mediq.dto.PrescriptionResponse;

import java.util.List;

public interface PrescriptionService {

    PrescriptionResponse createPrescription(CreatePrescriptionRequest request);

    PrescriptionResponse getPrescriptionById(Long prescriptionId);

    PrescriptionResponse getPrescriptionByCode(String prescriptionCode);

    PrescriptionResponse getPrescriptionByConsultationId(Long consultationId);

    List<PrescriptionResponse> getPatientPrescriptions(Long patientId);

    List<PrescriptionResponse> getPrescriptionsByStatus(Long campId, PrescriptionStatus status);

    byte[] generatePrescriptionPdf(Long prescriptionId);
}

File: mediq-frontend/src/services/prescriptionService.js
import api from './api';

export const prescriptionService = {
  createPrescription: async (data) => api.post('/prescriptions', data),
  getPrescriptionById: async (id) => api.get(`/prescriptions/${id}`),
  getPrescriptionByCode: async (code) => api.get(`/prescriptions/code/${code}`),
  getPrescriptionByConsultationId: async (consultationId) => api.get(`/prescriptions/consultation/${consultationId}`),
  getPatientPrescriptions: async (patientId) => api.get(`/prescriptions/patient/${patientId}`),
  getPrescriptionsByStatus: async (campId, status) => api.get(`/prescriptions/status/${campId}`, { params: { status } }),
  getPrescriptionPdfUrl: (id) => `/api/v1/prescriptions/${id}/pdf`,
  downloadPrescriptionPdf: async (id) => api.get(`/prescriptions/${id}/pdf`, { responseType: 'blob' }),
};

export default prescriptionService;
